File: src/PlatynUI/technology/uiautomation/core/displaydevice.py
# SPDX-FileCopyrightText: 2024 Daniel Biehl <daniel.biehl@imbus.de>
#
# SPDX-License-Identifier: Apache-2.0

from typing import Optional

from PlatynUI.core.types import Rect
from PlatynUI.ui.core.devices.basedisplaydevice import BaseDisplayDevice

from .loader import DotNetInterface


class UiaDisplayDevice(BaseDisplayDevice):
    @property
    def name(self) -> str:
        return "Windows"

    @property
    def bounding_rectangle(self) -> Rect:
        r = DotNetInterface.display_device().GetBoundingRectangle()
        return Rect(r.X, r.Y, r.Width, r.Height)

    def highlight_rect(self, rect: Rect, time: float) -> None:
        DotNetInterface.display_device().HighlightRect(rect.left, rect.top, rect.width, rect.height, time)

    def get_screenshot(self, rect: Rect, format: str, quality: Optional[int]) -> bytearray:
        raise NotImplementedError

File: src/PlatynUI.Runtime/Core/DisplayDevice.cs
// SPDX-FileCopyrightText: 2024 Daniel Biehl <daniel.biehl@imbus.de>
//
// SPDX-License-Identifier: Apache-2.0

namespace PlatynUI.Runtime.Core;

public interface IDisplayDevice
{
    Rect GetBoundingRectangle();
    void HighlightRect(double x, double y, double width, double height, double time);
}
